mwdb/web/src/components/ShowObject/Views/RelationsBox.js

```diff
@@ -0,0 +1,130 @@
+                <th>parent</th>
+                <td>
+                    <span>
+                        <ObjectLink {...parent}
+                                    diffWith={parent.type === "text_blob" && (array[index + 1] || {}).id}
+                                    inline/>
+                    </span>
+                    <span className="ml-2">
+                        <ActionCopyToClipboard text={parent.id} tooltipMessage="Copy sha256 to clipboard"/>
+                    </span>
+                </td>
+                <td>
+                    <TagList tags={parent.tags}/>
+                </td>
+            </tr>
+        );
+
+    const children = (props.children || [])
+        .map((child, index, array) =>
+            <tr key={`child-${child.id}`} className="flickerable">
+                <th>child</th>
+                <td>
+                    <span>
+                        <ObjectLink {...child }
+                                    diffWith={child.type === "text_blob" && (array[index + 1] || {}).id}
+                                    inline/>
+                    </span>
+                    <span className="ml-2">
+                        <ActionCopyToClipboard text={child.id} tooltipMessage="Copy sha256 to clipboard"/>
+                    </span>
+                </td>
+                <td>
+                    <TagList tags={child.tags}/>
+                </td>
+            </tr>
+        );
+    return (
+        <div className="card card-default">
+            <div className="card-header">
+                { props.header || "Relations" }
+                <Link to="#" className="float-right"
+                      onClick={(ev) => {
+                          ev.preventDefault();
+                          setAttributeAddModalOpen(true);
+                          setModalError("");
+                      }}>
+                        <FontAwesomeIcon icon="plus" pull="left" size="1x"/>
+                        Add
+                </Link>
+            </div>
+            {
+                parents.length + children.length > 0 ?
+                <table className="table table-striped table-bordered table-hover" id="rel_table">
+                    <tbody id="rel_body">
+                    { parents }
+                    { children }
+                    </tbody>
+                </table> : <div className="card-body text-muted">No relations to display</div>
+            }
+            <RelationsAddModal isOpen={isAttributeAddModalOpen}
+                               error={modalError}
+                               onSubmit={addObjectRelations}
+                               onError={setModalError}
+                               onRequestModalClose={() => setAttributeAddModalOpen(false)}
+            />
+        </div>
+    );
+}
+
+export default function MultiRelationsBox(props) {
+    const context = useContext(ObjectContext);
+
+    let TypedRelationsBox = (props) => {
```

Comment:
I wonder if we should move it to top-level scope instead of keeping this component inside the function.

When function is declared inside the component, it will create new function instance every time the MultiRelationsBox is rendered. This may cause premature unmount/mount of TypedRelationsBox because React doesn't know that we doesn't put anything new there.

Alternative solution is useCallback but it sounds a bit dirty to keep a helper component there 🤔